### DISCUSSION TITLE
Ico support

### DISCUSSION THREAD

**Discussion by kdh8219** [spacedriveapp/spacedrive#1968] [Feature Request]
```
2024-01-21T12:47:09.408831Z ERROR sd_core::object::media::thumbnail::process: core/src/object/media/thumbnail/process.rs:219: Failed to generate thumbnail for indexed location: SdImages {
    path: "/.../vw-data/icon_cache/www.lottecinema.co.kr.png",
    error: Image(
        Decoding(
            DecodingError {
                format: Exact(
                    Ico,
                ),
                underlying: Some(
                    PngNotRgba,
                ),
            },
        ),
    ),
}
```
add ico support

**Comment by brxken128** [comment]
That appears to be a `.ico` file possibly renamed to `.png`, judging from the extension? ICOs do have support to store an entire PNG image within them, but they must have their own ARGB (or RGBA) header and be self-contained ([source](https://en.wikipedia.org/wiki/ICO_(file_format)#PNG_format)), which is likely why the error message is `PngNotRgba`.

If the image is actually a full `.ico` and was renamed, to a different extension, the _format_ of the internal image hasn't changed and changing extension would be incorrect. I believe we could go with the magic-byte reading approach to correctly label and load images here, and I think it'd be beneficial to do that for all images in order to prevent mix-ups like this from happening.

We do have support for `.ico` images, the issue here is it's been labelled as a `.png` (I think), and our thumbnail generation relies on the image extension as opposed to the magic bytes and the data within.
